<a href="https://colab.research.google.com/github/sdasmaha/Ml_Engineers_445/blob/master/Heart_Notebook_Exercise1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

 Notebook Excercise #1 - SRINJOY DASMAHAPATRA

 IDE => Google Colab


#Importing Libraries

In [ ]:
import numpy as np
import pandas as pd

#Mounting the Google Drive


In [ ]:
from google.colab import drive
drive.mount("/content/gdrive")

Mounted at /content/gdrive


#Loading the Dataset

In [ ]:
heart_df = pd.read_csv('/content/gdrive/My Drive/Colab Notebooks/BAJWA-ML/heart_failure_dataset.csv')

# *1.1*

**A**

In [ ]:
print(heart_df.shape) 

(299, 13)


In [ ]:
print(heart_df.axes)

[RangeIndex(start=0, stop=299, step=1), Index(['age', 'anaemia', 'creatinine_phosphokinase', 'diabetes',
       'ejection_fraction', 'high_blood_pressure', 'platelets',
       'serum_creatinine', 'serum_sodium', 'sex', 'smoking', 'time',
       'DEATH_EVENT'],
      dtype='object')]


In [ ]:
print(heart_df.dtypes)

age                         float64
anaemia                       int64
creatinine_phosphokinase      int64
diabetes                      int64
ejection_fraction             int64
high_blood_pressure           int64
platelets                   float64
serum_creatinine            float64
serum_sodium                  int64
sex                           int64
smoking                       int64
time                          int64
DEATH_EVENT                   int64
dtype: object


**B**

In [ ]:
heart_df.head(10)

age  anaemia  creatinine_phosphokinase  ...  smoking  time  DEATH_EVENT
0  75.0        0                       582  ...        0     4            1
1  55.0        0                      7861  ...        0     6            1
2  65.0        0                       146  ...        1     7            1
3  50.0        1                       111  ...        0     7            1
4  65.0        1                       160  ...        0     8            1
5  90.0        1                        47  ...        1     8            1
6  75.0        1                       246  ...        0    10            1
7  60.0        1                       315  ...        1    10            1
8  65.0        0                       157  ...        0    10            1
9  80.0        1                       123  ...        1    10            1

[10 rows x 13 columns]

**C**
    Each row of heart_df dataframe is termed called a data sample associated with an instance.

**D**    Based on the knowledge of the dataset and its stated usage, we are dealing with supervised problem. 

Since in a supervised learning model, the algorithm learns on a labeled dataset, providing an answer key that the algorithm can use to evaluate its accuaracy on training data. 

On the contrary the unsupervised model works on unlabeled data that the algorithm tries to make sense of by extracting features and patterns on its own.


**E** The independent variables in this case are  labels- 

1. age
2. anaemia
3. creatinine_phosphokinase
4. diabetes
5. ejection_fraction
6. high_blood_pressure
7. platelets
8.	serum_creatinine
8. serum_sodium
9. sex
10. smoking
11. time

This is so because these variable/labels embedded in the algorithm 
increase or decrease the probability/likhehood of the event(*DEATH_EVENT*).


**F** The dependent variable in this case is the label *DEATH_EVENT* since it is the conclusive/final event or variable that predicts the likelihood of death.

**G** Here the categorical variables are -


1. anaemia
2. diabetes 
3. high_blood_pressure
4. sex
5. smoking
6. DEATH_EVENT


This is so because a categorical variable is a variable that can take on one of a limited, and usually fixed, number of possible values, assigning each individual or other unit of observation to a particular group or nominal category on the basis of some qualitative property.



**H** The categorical variables follow *L Encoding* where each label is mapped to a binary number.

In [ ]:
count1 = heart_df['DEATH_EVENT'].value_counts() 
print(count1)

0    203
1     96
Name: DEATH_EVENT, dtype: int64


**I** 


1.   Samples corresponding to Deceased Patients : 96
2.   Samples corresponding to Remaining Patients : 203



In [ ]:
count2 = heart_df['sex'].value_counts() 
print(count2)

NameError: ignored

**J** 

1.   Samples corresponding to Women Patients : 105
2.   Samples corresponding to Male Patients : 194


In [ ]:
count3 = heart_df['smoking'].value_counts() 
print(count3)

0    203
1     96
Name: smoking, dtype: int64


**K** 

1.   Samples corresponding to Smokers  : 96
2.   Samples corresponding to Non-Smokers : 203


#*1.2*

Side Notes

1.   Positive Correlation = A positive relationship exists i.e. when the indepenedent variable increases in value, so does the dependent variable

2.  Negative Correlation = A negative relationship exists i.e. when the independent variable increases in value, the dependent variable falls in value



In [ ]:
heart_df.corr()

age   anaemia  ...      time  DEATH_EVENT
age                       1.000000  0.088006  ... -0.224068     0.253729
anaemia                   0.088006  1.000000  ... -0.141414     0.066270
creatinine_phosphokinase -0.081584 -0.190741  ... -0.009346     0.062728
diabetes                 -0.101012 -0.012729  ...  0.033726    -0.001943
ejection_fraction         0.060098  0.031557  ...  0.041729    -0.268603
high_blood_pressure       0.093289  0.038182  ... -0.196439     0.079351
platelets                -0.052354 -0.043786  ...  0.010514    -0.049139
serum_creatinine          0.159187  0.052174  ... -0.149315     0.294278
serum_sodium             -0.045966  0.041882  ...  0.087640    -0.195204
sex                       0.065430 -0.094769  ... -0.015608    -0.004316
smoking                   0.018668 -0.107290  ... -0.022839    -0.012623
time                     -0.224068 -0.141414  ...  1.000000    -0.526964
DEATH_EVENT               0.253729  0.066270  ... -0.526964     1.000000

[13 rows x 13 columns]

**A** The two variables that are the most positively correlated with the *DEATH-EVENT* are -

1.   serum_creatinine (0.294278)
2.   age (0.253729)





**B** The two variables that are the most negatively correlated with the *DEATH-EVENT* are -

1.   time (-0.526964	)
2.   ejection_fraction (-0.268603	)






**C** The answer to this question can be understood by general intuition and research conducted in the past. With increasing age, DEATH_EVENT is likely to increase.  

**D** Past Research has shown with decrease in ejection fraction(Percentage of blood leaving the heart at each contraction) there is an increase in the likelihood of Death (DEATH_EVENT).

#*1.3*


1.Checking for missing values.


In [ ]:
heart_df.isnull().sum()

age                         0
anaemia                     0
creatinine_phosphokinase    0
diabetes                    0
ejection_fraction           0
high_blood_pressure         0
platelets                   0
serum_creatinine            0
serum_sodium                0
sex                         0
smoking                     0
time                        0
DEATH_EVENT                 0
dtype: int64

We can see here that we are not dealing with missing values.

2. Checking for Inconsistent values. For this we have to assess each columns and categorize them first into their data types and then search for inconsistency.


1.   Age = Any negative value or age greater than 110 might be inconsistent
2.   List item



We Can Clearly see that there is no negative values!